docs/manifest.md

```diff
@@ -78,6 +78,77 @@ examples/foo/bar:
     - if: IDF_TARGET == "esp32s2"
 ```
 
+## Enhanced YAML Syntax
+
+### Reuse Lists
+
+To extend and remove elements from a list, you can use the `+` and `-` postfixes respectively to modify existing arrays. The order of calculation is always `+` first, followed by `-`.
+
+#### Array Elements as Strings
+
+The following YAML code demonstrates how to extend and remove elements from a list of strings:
+
+```yaml
+.base_depends_components: &base-depends-components
+  depends_components:
+    - esp_hw_support
+    - esp_rom
+    - esp_wifi
+
+examples/wifi/coexist:
+  <<: *base-depends-components
+  depends_components+:
+    - esp_coex
+  depends_components-:
+    - esp_rom
+```
+
+After interpretation, the resulting YAML will be:
+
+```yaml
+examples/wifi/coexist:
+  depends_components:
+    - esp_hw_support
+    - esp_wifi
+    - esp_coex
+```
+
+This means that the `esp_rom` element is removed, and the `esp_coex` element is added to the `depends_components` list.
+
+#### Array Elements as Dictionaries
+
+In addition to extending and removing elements from a list of strings, you can also perform these operations on a list of dictionaries. The matching is done based on the `if` key. Here's an example:
+
+```yaml
+.base: &base
+  enable:
+    - if: IDF_VERSION == "5.2.0"
+    - if: IDF_VERSION == "5.3.0"
+
+foo:
+  <<: *base
+  enable+:
+    - if: IDF_VERSION == "5.2.0"
+      temp: true  # <-- This line will override 'enable'
+    - if: IDF_VERSION == "5.4.0"
+      reason: bar
+```
+
+After interpretation, the resulting YAML will be:
+
+```yaml
+foo:
+  enable:
+  - if: IDF_VERSION == "5.3.0"
+  - if: IDF_VERSION == "5.2.0"
+    temp: true
+  - if: IDF_VERSION == "5.4.0"
+    reason: bar
+```
+
+In this case, the `enable` list is extended with the new `if` statement and `reason` dictionary. It's important to note that the line with `temp: true` will override the existing `enable` entry because the `if` statement matches an existing one.
```

Comment:
This line is not accurate. not the "line with `temp:true` will override, but the `if` dictionary defined later will override the earlier one.

Reply:
changed to 

It's important to note that the `if` dictionary defined in the `+` postfix will override the earlier one when the `if` statement matches.